test/src/actions/AddDGDiffusion.C

```diff
@@ -0,0 +1,65 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "AddDGDiffusion.h"
+#include "Factory.h"
+#include "InputParameters.h"
+#include "FEProblemBase.h"
+
+registerMooseAction("MooseTestApp", AddDGDiffusion, "add_dg_kernel");
+
+template <>
+InputParameters
+validParams<AddDGDiffusion>()
+{
+  auto params = validParams<Action>();
+  params.addRequiredParam<NonlinearVariableName>("variable",
+                                                 "The variable on which the dgkernel will act");
+  params.addParam<Real>("sigma", "sigma");
+  params.addParam<Real>("epsilon", "epsilon");
+  params.addParam<MaterialPropertyName>(
+      "diff", "The diffusion (or thermal conductivity or viscosity) coefficient.");
+  params.addParam<VariableName>("coupled_var",
+                                "The coupled variable that will determine the diffusion "
+                                "rate from the DGCoupledConvection dgkernel");
+  MooseEnum kernels_to_add("BOTH REGULAR COUPLED", "BOTH");
+  params.addParam<MooseEnum>(
+      "kernels_to_add", kernels_to_add, "What dgkernels to add from this action");
+
```

Comment:
So this action does not need to worry about adding relationship managers even it is adding DG kernels?

Reply:
It should, but what we are testing here is whether the `CouplingFunctorCheckAction` can save developers who **did not** properly add ghosting in their actions. The `CouplingFunctorCheckAction` is capable of saving developers who forgot algebraic or coupling ghosting, but it cannot save them if they/we forgot geometric ghosting

Reply:
So it's worth discussing (@permcody @friedmud ) whether we want to keep default geometric ghosting, which has been happening behind the scenes even after #13145

Reply:
I do not think geometric ghosting is costly. Only because of using PETSc, we do not it in some circumstances if I understand correctly. In general, we should keep geometric ghosting in my opinion. As a developer, it will be nice if I do not need to worry about algebraic and coupling ghosting.

Reply:
I think we should always minimize ghosting. No need to keep something that isn't used. @YaqiWang - what about a a relatively simple equation system with adaptivity turned on? Adaptivity can be very expensive and with ghosting on, you increase communication at every step.

Reply:
Adaptivity requires ghosting though...

Reply:
I do not think geometric ghosting requires any communication during the solve. It is about how we handle the mesh distribution. It is done where mesh is setup for use.

Reply:
and then with each adaptive step...

Reply:
Right. But geometric ghosting costs much less than others both in CPU time and memory. It is not worthwhile to spend much effort to optimizing it probably.

Reply:
I would agree with you if it were a question of optimization. It's more a question of correctness. If it's as simple as turning something off (which we thought was off anyway), and things still work - I don't see why not.

Reply:
It would be good to conjure up a problem where geometric ghosting is unnecessary and where we think it might be expensive. I'm struggling to think of something that would satisfy the latter condition. In a regular simulation case the mesh should only run geometric ghosting functors and delete remote elements once, when `prepare_for_use` is called. Adaptivity would run the functors every time step, but then those functors are actually needed.

The problem with ***not*** having default geometric ghosting really only arises for a small sub-set of MOOSE users...those users who generate all their `MooseObjects` through custom actions, thus avoiding the `MooseObjectAction::addRelationsipManagers` call. For this small sub-set of users, they must either have default geometric ghosting or they must override `addRelationshipManagers(input_rm_type)` in their custom action.

Reply:
If this is a problem, how about let application call something to indicate having the default geometric ghosting? For example, we can call something in RattlesnakeApp.

Reply:
I'm on board for something like that